lib/feeb/db/repo.ex

```diff
@@ -133,6 +139,19 @@ defmodule Feeb.DB.Repo do
     end
   end
 
+  def handle_call({:mgt_connection_released}, {caller_pid, _}, state) do
+    # Make sure only the Repo.Manager can send mgt signals to the Repo.
+    assert_release_signal_from_manager!(state.manager_pid, caller_pid)
+
+    if not is_nil(state.transaction_id) do
+      Logger.info("Connection released forcibly; rolling back transaction #{state.transaction_id}")
+      :ok = SQLite.exec(state.conn, "ROLLBACK")
+    end
+
+    # Reset the GenServer state so we are ready to serve a new request
+    {:reply, :ok, %{state | transaction_id: nil}}
+  end
```

Comment:
> there may be a race condition if a transaction is committed or rolled back just before the check for state.transaction_id

That's not true. If we are handling the `:mgt_connection_released` call, then the GenServer is unable to process any other messages at the same time.

If it processes a `:commit` message immediately after, it will fail because you can't `COMMIT` with `transaction_id=nil`.